primitives/state-machine/src/changes_trie/build.rs

```diff
@@ -132,50 +132,52 @@ fn prepare_extrinsics_input_inner<'a, B, H, Number>(
 	block: &Number,
 	overlay: &'a OverlayedChanges,
 	child_info: Option<ChildInfo>,
-	changes: impl Iterator<Item=(&'a StorageKey, &'a OverlayedValue)>
+	mut changes: impl Iterator<Item=(&'a StorageKey, &'a OverlayedValue)>
 ) -> Result<impl Iterator<Item=InputPair<Number>> + 'a, String>
 	where
 		B: Backend<H>,
 		H: Hasher,
 		Number: BlockNumber,
 {
 	changes
-		.filter(|( _, v)| v.extrinsics().next().is_some())
```

Comment:
If you change this to a `filter_map`, you almost need no changes below.

Reply:
well it avoid a tab, will change (easier to review, even if I like it better without the filter map, I am really waiting for rustc "Tracking issue for methods converting `bool` to `Option<T>` #64260" to get stable).

Reply:
Oh I remember, 'extrinsics' function is not a getter, it have an associated cost, that is the main reason (even if optimized by the compilator, this way of putting it makes it clear that we don't want to call the function multiple times, but I can write it with combinator too).